### PR TITLE
Dast support api changes

### PR DESCRIPTION
**Pull request by krchandak**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java [hunk 1]
```diff
@@ -280,6 +291,236 @@ public void saveReleaseSettingsForWorkflowDrivenScan(String userSelectedRelease,
...
+                    dastPostmanScanSetupReqModel.setCollectionFileIds(ConvertStringtoIntArr(postmanIdCollection));
+                }
+
+            PutDastScanSetupResponse response = dynamicController.putDastPostmanScanSettings(Integer.parseInt(userSelectedRelease),
+                    dastPostmanScanSetupReqModel);
+            if (response.isSuccess && response.errors == null) {
+                System.out.println("Successfully saved settings for release id = " + userSelectedRelease);
+
+            } else {
+                throw new Exception(String.format("Failed to save scan settings for release id %d", Integer.parseInt(userSelectedRelease)));
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java:509]: please move error strings to FodGlobal Constants.

File: src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java [hunk 2]
```diff
@@ -280,6 +291,236 @@ public void saveReleaseSettingsForWorkflowDrivenScan(String userSelectedRelease,
...
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException(String.format("Failed to save scan settings for release id %d", Integer.parseInt(userSelectedRelease)), e);
+        }
+    }
+    public int[] ConvertStringtoIntArr(String fileIds) {
+        String[] postmanIds = fileIds.split(",");
+        int[] postmanIdArr = new int[postmanIds.length];
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java:517]: Should we check for the null here before accessing the length property

File: src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java [hunk 3]
```diff
@@ -16,9 +16,7 @@
 import java.io.PrintStream;
 import java.util.ArrayList;
 
-
-import static org.jenkinsci.plugins.fodupload.Config.FodGlobalConstants.FodDastApiConstants.DastWebSiteScanPutApi;
-import static org.jenkinsci.plugins.fodupload.Config.FodGlobalConstants.FodDastApiConstants.DastWorkflowScanPutApi;
+import static org.jenkinsci.plugins.fodupload.Config.FodGlobalConstants.FodDastApiConstants.*;
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java:19]: Good change,

File: src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java [hunk 4]
```diff
@@ -90,6 +88,22 @@ public PatchDastFileUploadResponse PatchDynamicScan(PatchDastScanFileUploadReq r
                     temp = File.createTempFile("worflowdriven", requestModel.releaseId);
                     fileName = "WorkflowDriven.webmacro";
                     break;
+                case OpenAPIDefinition:
+                    temp = File.createTempFile("openAPIDefinition", requestModel.releaseId);
+                    fileName = "OpenAPIDefinition.json";
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java:93]: File Name strings to FodGlobalConstants.

File: src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java [hunk 5]
```diff
@@ -90,6 +88,22 @@ public PatchDastFileUploadResponse PatchDynamicScan(PatchDastScanFileUploadReq r
...
+                    fileName = "GRPCDefinition.proto";
+                    break;
+                case GraphQLDefinition:
+                    temp = File.createTempFile("graphQLDefinition", requestModel.releaseId);
+                    fileName = "GraphQLDefinition.json";
+                    break;
+                case PostmanCollection:
+                    temp = File.createTempFile("postmanCollection", requestModel.releaseId);
+                    fileName = "PostmanCollection.json";
+                    break;
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java:106]: please Add try catch block to delete the temp file.   in catch ignore the error.

File: src/main/java/org/jenkinsci/plugins/fodupload/models/DastScanJobModel.java [hunk 6]
```diff
@@ -30,8 +30,23 @@ public class DastScanJobModel {
     String entitlementFrequencyId;
     String entitlementFrequencyType;
     String userSelectedEntitlement;
-
     String networkAuthType;
+    String selectedApi;
```
Comment (pvivekdev) [inline src/main/java/org/jenkinsci/plugins/fodupload/models/DastScanJobModel.java:34]: Can we have one more Overload Constructor to cover for API so We don't need to invoke Ctor with null values passed for website /workflow driven

File: src/main/webapp/ScanSettings.js [hunk 7]
```diff
@@ -47,7 +47,7 @@ class ScanSettings {
         let at = this.assessments ? this.assessments[atval] : null;
 
         entsel.find('option,optgroup').remove();
-
+        debugger;
```
Comment (pvivekdev) [inline src/main/webapp/ScanSettings.js:50]: please remove all debugger. I have missed removing  it too :)

File: src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java (hunk 2 above)
Comment (krchandak) [reply src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java:517]: This is controller part. I belive all checks inclusing this should happen before reaching to it. Should we do it again here.?

File: src/main/java/org/jenkinsci/plugins/fodupload/models/DastScanJobModel.java (hunk 6 above)
Comment (krchandak) [reply src/main/java/org/jenkinsci/plugins/fodupload/models/DastScanJobModel.java:34]: Sure, will do it in another PR

File: src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java (hunk 5 above)
Comment (krchandak) [reply src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java:106]: As discussed it exist

File: src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java (hunk 3 above)
Comment (krchandak) [reply src/main/java/org/jenkinsci/plugins/fodupload/controllers/DastScanController.java]: Thanks

File: src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java (hunk 1 above)
Comment (krchandak) [reply src/main/java/org/jenkinsci/plugins/fodupload/DastScanSharedBuildStep.java:509]: SUre, I have done it but need to use it

**Pull request merged**